Sources/Workspace/Diagnostics.swift

```diff
@@ -42,3 +42,14 @@ extension ManifestParseError: DiagnosticDataConvertible {
         }
     }
 }
+
+extension DiagnosticsEngine {
+    func wrap<T>(_ closure: () throws -> T) -> T? {
```

Comment:
Nice! I think this can be moved into Utility:
https://github.com/apple/swift-package-manager/blob/master/Sources/Utility/Diagnostics.swift#L50